test/pkg/integration/integration.go

```diff
@@ -208,6 +209,11 @@ func (r *RpcClient) Call(serviceMethod string, args any, reply any) error {
 	for i := 0; i < connectFailureMaxTries; i++ {
 		if r != nil {
 			if err = r.client.Call(serviceMethod, args, reply); err != nil {
+				log.Warnf("rpc call %s failed (attempt %d): %v", serviceMethod, i, err)
+				if i == connectFailureMaxTries-1 {
+					return err
+				}
```

Comment:
on the final attempt, return the RPC error, otherwise the error gets swallowed below